locallibrary/catalog/forms.py

```diff
@@ -0,0 +1,21 @@
+import datetime
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
```

Comment:
Định nghĩa các trường có trong form
ngoài DateField còn có nhiều field khác dc hỗ trợ như : BooleanField, CharField, ChoiceField, TypedChoiceField, DateField, DateTimeField, DecimalField, DurationField, EmailField, FileField, FilePathField, FloatField, ImageField, IntegerField, GenericIPAddressField, MultipleChoiceField, TypedMultipleChoiceField, NullBooleanField, RegexField, SlugField, TimeField, URLField, UUIDField, ComboField, MultiValueField, SplitDateTimeField, ModelMultipleChoiceField, ModelChoiceField.

Field có các thuộc tính như:
+ required: nếu là True thì là bắt buộc nhập. Mặc định required là true. nếu muốn để giá trị trống thì set required=False
+ label : Label hiển thị. nếu k set nó sẽ lấy tên trường rồi in hoa chữ cái đầu, bỏ dấu _
vd: ở đây sẽ là Renewal date
+ initial : set giá trị mặc định ban đầu
+ help_text : dòng text hiển thị thêm mô tả field đó
+ error_messages : Danh sách các error message cho field. 
+ ....